Document `wasm-bindgen` without `worker-build` (#9389)
cloudflare/cloudflare-docs: content/workers/platform/webassembly/rust.md

This interoperability is achieved using [`wasm-bindgen`](https://rustwasm.github.io/wasm-bindgen/), which provides the glue code needed to import runtime APIs to, and export event handlers from, the Wasm module. `wasm-bindgen` also provides [`js-sys`](https://docs.rs/js-sys/latest/js_sys/), which implements types for interacting with JavaScript objects. In practice, this is an implementation detail, as `workers-rs`'s API handles conversion to and from JavaScript objects, and interaction with imported JavaScript runtime APIs for you.

{{<Aside type="note">}}

> If you are using `wasm-bindgen` without `workers-rs` / `worker-build`, then you will need to patch the JavaScript that it emits.
> This is because when you import a `wasm` file in Workers, you get a `WebAssembly.Module` instead of a `WebAssembly.Instance` for
> performance and security reasons.

Reviewer: ```suggestion
If you are using `wasm-bindgen` without `workers-rs` / `worker-build`, then you will need to patch the JavaScript that it emits. This is because when you import a `wasm` file in Workers, you get a `WebAssembly.Module` instead of a `WebAssembly.Instance` for performance and security reasons.
```

> 1. Run `wasm-pack build --target bundler` as you normally would.

Reviewer: Can we introduce what these steps are doing before step 1?

"To ___< accomplish said task > _____:"

Author: Wouldn't that be "If you are using wasm-bindgen without workers-rs / worker-build, then you will need to patch the JavaScript that it emits"

> 1. Run `wasm-pack build --target bundler` as you normally would.

Reviewer: ```suggestion
To patch the JavaScript that `wasm-bindgen` emits:

1. Run `wasm-pack build --target bundler` as you normally would.
```

> 2. Patch the JavaScript file that it produces (here we assume the file is called `mywasmlib.js`):

Reviewer: ```suggestion
2. Patch the JavaScript file that it produces (the following code block assumes the file is called `mywasmlib.js`):
```

```js
import * as imports from "./mywasmlib_bg.js";
 
// switch between both syntax for node and for workerd
import wkmod from "./mywasmlib_bg.wasm";
import * as nodemod from "./mywasmlib_bg.wasm";
if ((typeof process !== 'undefined') && (process.release.name === 'node')) {
    imports.__wbg_set_wasm(nodemod);
} else {
    const instance = new WebAssembly.Instance(wkmod, { "./mywasmlib_bg.js": imports });
    imports.__wbg_set_wasm(instance.exports);
}
 
export * from "./mywasmlib_bg.js";

> ```

Author: Thanks, I figured I would keep this for portability.

> 3. In your Worker script, import the function and use it directly:

Reviewer: ```suggestion
3. In your Worker entrypoint, import the function and use it directly:
```
We do not use "script" when referring to Worker files anymore. It should be 'entrypoint' or if writing a tutorial, it should be the exact filename.
